internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -627,6 +623,15 @@ func setCondition(condition hivev1.ClusterDeploymentCondition, conditions *[]hiv
 	}
 }
 
+func clearClusterAPIErrors(cluster *hivev1.ClusterDeployment) {
+	conditions := &cluster.Status.Conditions
+	for i := len(*conditions) - 1; i >= 0; i-- {
+		if (*conditions)[i].Reason == AgentPlatformError {
+			*conditions = append((*conditions)[:i], (*conditions)[i+1:]...)
```

Comment:
can you please explain this?

Reply:
here is just a removal of the condition in index i